styles/scss/mixins/_underline-text.scss

```diff
@@ -0,0 +1,109 @@
+///  This mixin has 2 possible way of using it.
+///  First is to include in the class you want to hover over while text
+///  you want to underline needs to have attached class .underline-text
+///  or you can send your custom class name via parameter $element.
+///  Also you have to set $wrapper parameter to true.
+///  Second is to just include it in whichever element you want. In this
+///  case, parameter $wrapper should stay unchanged.
+///  Other parameters are optional and they allow you customize your transition.
+///
+/// @access public
+/// @author Karlo Volf
+/// @param {percentage} $thickness [10%] - Thickness of the line in percentage.
+/// @param {seconds} $duration [.5s] - How long animation lasts.
+/// @param {timing-function} $timing-function [cubic-bezier(.79, .01, .22, .99)] - Execution speed function.
+/// @param {hex} $color [currentColor] - Color of the line.
+/// @param {boolean} $wrapper [false] - If wrapper is used as hover reference.
+/// @param {string} $element [.underline-text] - Element on which you want underline.
+/// @param {string} $state [hover] - The state in which it becomes underline element.
+/// @example
+/// .test {
+///   @include underline-text();
+/// }
+/// .test2 {
+///   @include underline-text(10, .7s, ease-in, #132031, true, a.link, hover);
+/// }
```

Comment:
Maybe add another example of mixing usage. This way some mixin arguments can be used by default and only specific ones will be overridden.
```
@include underline-text(
  $thickness: 10,
  $duration: .7s
  ...
);
```